sdk/nodejs/cmd/run/run.ts

```diff
@@ -487,7 +487,8 @@ ${defaultMessage}`,
     };
 
     // Construct a `Stack` resource to represent the outputs of the program.
-    const stackOutputs = stack.runInPulumiStack(runProgram);
+    const stackOutputs = await stack.runInPulumiStack(runProgram);
+    await settings.disconnect()
```

Comment:
We need to disconnect here so that the callback server can be shutdown, otherwise it leaves the nodejs event loop forever running and we never hit `process.on("exit", settings.disconnectSync);`